components/infobox/wikis/valorant/infobox_team_custom.lua

```diff
@@ -17,40 +18,37 @@ local Team = Lua.import('Module:Infobox/Team', {requireDevIfEnabled = true})
 local Widgets = require('Module:Infobox/Widget/All')
 local Cell = Widgets.Cell
 
-local CustomTeam = Class.new()
+local CustomTeam = Class.new(Team)
 local CustomInjector = Class.new(Injector)
 
-local _team
-
 function CustomTeam.run(frame)
-	local team = Team(frame)
-	_team = team
-	team.createWidgetInjector = CustomTeam.createWidgetInjector
-	team.createBottomContent = CustomTeam.createBottomContent
-	team.addToLpdb = CustomTeam.addToLpdb
+	local team = CustomTeam(frame)
+	team:setWidgetInjector(CustomInjector(team))
 	return team:createInfobox()
 end
 
-function CustomTeam:createWidgetInjector()
-	return CustomInjector()
-end
-
 function CustomInjector:parse(id, widgets)
+	local args = self.caller.args
 	if id == 'staff' then
-		table.insert(widgets, Cell{
-			name = 'In-Game Leader',
-			content = {_team.args.igl}
-		})
+		return {
+			Cell{name = 'In-Game Leader', content = {args.igl}}
+		}
+	end
+
+	if id == 'customcontent' then
+		return Array.append(widgets,
```

Comment:
no need to append here
append makes sense if you add >1 widget
else table.insert is completely fine
or in the case of widgets being an empty table just `return {Cell{...}}`